src/main/java/com/uw/dao/TraineeDaoImpl.java

```diff
@@ -1,43 +1,179 @@
 import org.springframework.stereotype.Repository;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @Repository
 public class TraineeDaoImpl implements TraineeDao{
 
-    private Storage storage;
+    private SessionFactory sessionFactory;
 
     @Autowired
-    public void setStorage( Storage storage ){
-        this.storage = storage;
+    public void setSessionFactory( SessionFactory sessionFactory ){
+        this.sessionFactory = sessionFactory;
     }
 
     @Override
     public void create(Trainee trainee) {
-        long id = this.storage.getSize() + 1;
-        trainee.setUserId( id );
-        this.storage.save( trainee.getUserId(), trainee );
+        Session session = sessionFactory.openSession();
+        Transaction transaction = null;
+        try{
+            transaction = session.beginTransaction();
+            session.persist(trainee);
+            transaction.commit();
+        }catch( Exception e ){
+            transaction.rollback();
+        }finally{
+            session.close();
+        }
     }
 
     @Override
     public void update(Trainee trainee) {
-        this.storage.save( trainee.getUserId(), trainee );
+        Session session = sessionFactory.openSession();
+        Transaction transaction = null;
+        try{
+            transaction = session.beginTransaction();
+            session.merge(trainee);
+            transaction.commit();
+        }catch(Exception e){
+            transaction.rollback();
+        }finally{
+            session.close();
+        }
     }
 
     @Override
     public void delete(long id) {
-        this.storage.remove( id );
+        Session session = sessionFactory.openSession();
+        Transaction transaction = null;
+        try{
+            transaction = session.beginTransaction();
+            Trainee traineeToDelete = session.get(Trainee.class, id);
+            session.remove(traineeToDelete);
+            transaction.commit();
+        }catch(Exception e){
+            transaction.rollback();
+        }finally{
+            session.close();
+        }
     }
 
     @Override
     public Trainee selectProfile(long id) {
-        return ( Trainee ) this.storage.get( id );
+        Session session = this.sessionFactory.openSession();
+        Transaction transaction = null;
+        Trainee trainee = null;
+        try{
+            trainee = session.get (Trainee.class, id);
+        }catch ( Exception e){
+            e.printStackTrace();
+        }finally{
+            session.close();
+        }
+        return trainee;
+    }
+
+    @Override
+    public List<Trainee> findAll() {
+        Session session = sessionFactory.openSession();
+        List<Trainee> traineeList = null;
+
+        try {
+            // Crear un objeto Criteria para la clase de entidad User
+            Query query = session.createQuery("select t from Trainee t", Trainee.class);
+            traineeList = query.getResultList();
+        } catch (HibernateException e) {
```

Comment:
You could use better java syntax catch (HibernateException | NoResultException e) ...to make the code less verbose